test/test_c10d.py

```diff
@@ -3240,6 +3247,51 @@ def test_broadcast_coalesced_gloo_cpu(self):
         device = torch.device('cpu')
         self._test_broadcast_coalesced(process_group, device)
 
+class BackendLoadingTest(MultiProcessTestCase):
+    def setUp(self):
+        super(BackendLoadingTest, self).setUp()
+        self._fork_processes()
+
+    def _load_c10d_test_backend(self):
+        temp_dir = tempfile.mkdtemp()
+        src = "{}/cpp_extensions/cpp_c10d_extension.cpp".format(os.path.abspath(os.path.dirname(__file__)))
+        extension = torch.utils.cpp_extension.load(
+            name="pytorch_test",
+            sources=[src],
+            build_directory=temp_dir
+        )
+
+    @skip_if_rocm
+    def test_collective_apis(self):
+        import torch.utils.cpp_extension
+        try:
+            torch.utils.cpp_extension.verify_ninja_availability()
+        except RuntimeError:
+            print(CPP_EXTENSIONS_WARNING)
+            return 0
+        self._load_c10d_test_backend()
+
+        os.environ['WORLD_SIZE'] = '1'
+        os.environ['MASTER_ADDR'] = '127.0.0.1'
+        os.environ['MASTER_PORT'] = str(common.find_free_port())
+        os.environ['RANK'] = '0'
+
+        c10d.init_process_group(backend='test', init_method='env://', world_size=1, rank=0)
+        self.assertEqual(c10d.get_rank(), 0)
+        self.assertEqual(c10d.get_world_size(), 1)
+
+        process_group = c10d.distributed_c10d._get_default_group()
+        work = process_group.allreduce([torch.rand(1), torch.rand(1)])
+        work.wait()
+        self.assertTrue(work.is_completed())
+        self.assertTrue(work.is_success())
+
+        work = process_group.broadcast([torch.rand(1)])
+        work.wait()
+        self.assertTrue(work.is_completed())
+        self.assertTrue(work.is_success())
+
+        c10d.destroy_process_group()
```

Comment:
This file should only test the c10d API directly. For `torch.distributed` tests, see `test_distributed.py`.

Reply:
moved the logic to test_distributed.py